File: app/src/main/java/com/alorma/reconnect/ReconnectReceiver.java
package com.alorma.reconnect;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class ReconnectReceiver extends BroadcastReceiver {
  @Override
  public void onReceive(Context context, Intent intent) {
    Intent serviceIntent = new Intent(context, ReconnectListenerService.class);
    context.startService(serviceIntent);
  }
}

File: app/src/main/java/com/alorma/reconnect/ReconnectReceiver.kt
package com.alorma.reconnect

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import org.jetbrains.anko.startService

class ReconnectReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        context.startService<ReconnectListenerService>()
    }
}
